src/contracts/digital-twin/digital-twin.spec.ts

```diff
@@ -186,9 +258,261 @@ describe('DigitalTwin', function test() {
+      const pinnedHashesAfterDeactivation = await getPinnedFileHashes();
+      pinnedHashes.forEach((pinned) => {
+        expect(pinnedHashesAfterDeactivation.includes(pinned)).to.be.false;
+      });
+    });
+
+    it('can deactivate a created twin (without hash unpinning check)', async () => {
+      const localRuntime = await getRuntimeWithEnabledPinning(runtime);
+
+      const configWithTemplate = {
+        ...defaultConfig,
+        ...twinTemplate,
+      };
+      const twin = await DigitalTwin.create(localRuntime as DigitalTwinOptions, configWithTemplate);
+      const twinIdentity = (await twin.getDescription()).identity;
+      const twinContract = await localRuntime.contractLoader.loadContract(
+        'DigitalTwin',
+        await twin.getContractAddress(),
+      );
+
+      // Create sharing
+      const entries = await twin.getEntries();
+      const randomSecret = `super secret; ${Math.random()}`;
+      await localRuntime.sharing.addSharing(
+        entries.plugin1.value.config.address,
+        accounts[0],
+        accounts[1],
+        '*',
+        0,
+        randomSecret,
+        null,
+        false,
+        null,
+      );
+
+      await twin.deactivate();
+
+      // Check if container's owner == 0x0
+      // & consumers are removed
+      // & authority is removed
+      // & description is unpinned
+      // & sharing is unpinned & removed
+      let containerAddress;
+      let containerContract;
+      let containerOwner;
+      let containerAuthority;
+      let consumerCount;
+      let sharingAddress;
+      for (const entry of Object.keys(entries)) {
+        if (entries[entry].entryType === DigitalTwinEntryType.Container) {
+          containerAddress = entries[entry].value.config.address;
+          containerContract = await localRuntime.contractLoader.loadContract('DataContract', containerAddress);
+          containerOwner = await localRuntime.executor.executeContractCall(containerContract, 'owner');
+          containerAuthority = await localRuntime.executor.executeContractCall(containerContract, 'authority');
+          consumerCount = await localRuntime.executor.executeContractCall(containerContract, 'consumerCount');
+          sharingAddress = await localRuntime.executor.executeContractCall(containerContract, 'sharing');
+
+          expect(containerOwner).to.eq(nullAddress);
+          expect(containerAuthority).to.eq(nullAddress);
+          expect(consumerCount.toString()).to.eq('0'); // BigNumber
+          expect(sharingAddress).to.eq(nullBytes32);
+        }
+      }
+
+      // Check if twin's description is 0x0
+      // & twin's owner is 0x0
+      // & twin's authority is 0x0
+      // & twin's identity's owner is 0x0
+      const twinOwner = await localRuntime.executor.executeContractCall(
+        twinContract,
+        'owner',
+      );
+      const twinAuthority = await localRuntime.executor.executeContractCall(
+        twinContract,
+        'owner',
```

Comment:
owner checked twice? do you mean 'authority' ?
```suggestion
        'authority',
```